core/Plugin.php

```diff
@@ -451,6 +451,19 @@ public static function getPluginNameFromNamespace($namespaceOrClassName)
     }
 
     /**
+     * Override this method in your plugin class if you want your plugin to be loaded during tracking.
+     *
+     * Note: If you define your own dimension or handle a tracker event, your plugin will automatically
+     * be detected as a tracker plugin.
+     *
+     * @return bool
+     */
+    public function isTrackerPlugin()
```

Comment:
> So there will be both approaches possible?

Yes, but that's not the point. If #8329 is merged, it will be possible as a side effect (and the objects will be injected, which right now is not possible using the "component" approach). You can tell developers to do what you want them to do and not tell them they can do it through DI if you want.

> Just to be sure: If a plugin is deactivated, the config.php of such a deactivated plugin is not loaded right?

Yes.